sources/poly.cc

```diff
@@ -2408,7 +2408,14 @@ void poly::get_variables (PHEAD vector<WORD *> es, bool with_arghead, bool sort_
 				degrees.push_back(1);
 			}
 		}
-		else {		
+		// JD: Here we need to check for non-symbol/number terms in fast notation.
+		else if (*e < 0 && *e != -SNUMBER && *e != -SYMBOL) {
```

Comment:
`*e` cannot be `-SNUMBER/-SYMBOL` because it is in the else-if-clause.

Reply:
Yes you are right. It redundant conditions might well be optimized away but we might as well remove them. It is clear from the comment what is happening.